### PR TITLE
fix(cli): make owner flag optional.

### PR DESCRIPTION
**Pull request by arijitAD**
## Description

* Make `--owner` flag optional for `lease` subcommand
* Fetch the value of `--owner` from `--from` is it is not provided.

I have created a list for the usage of owner flags.
https://hackmd.io/@FyilpQ7sQ265a9xyqSxyLg/HyKrGkLpd
Fixes: #1309 

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/ovrclk/akash/blob/master/CONTRIBUTING.md))
- [ ] My issue has received approval from the maintainers or lead with the `design/approved` label

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/ovrclk/akash/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by troian** [comment]
sorry, i might not put enough details into the issue.
by optional i meant to not remove `owner` but rather do it this way
- `akash tx lease create --dseq=100 --from` # owner is pulled from `--from`
- `akash tx lease create --dseq=100 --owner=akash1.... --from` #owner is pulled from `--owner`

**Comment by arijitAD** [comment]
@troian I have done it for `lease` command. Let me know if this seems correct and if this needs to be done for other commands.

File: x/deployment/client/cli/tx.go [hunk 1]
```diff
@@ -274,7 +274,7 @@ func cmdGroupClose(_ string) *cobra.Command {
 
 	flags.AddTxFlagsToCmd(cmd)
 	AddGroupIDFlags(cmd.Flags())
-	MarkReqGroupIDFlags(cmd, DeploymentIDOptionNoOwner(true))
```
Comment (troian) [inline x/deployment/client/cli/tx.go]: `--owner` should be optional so original `MarkReqGroupIDFlags(cmd, DeploymentIDOptionNoOwner(true))` is legit

**Comment by troian** [comment]
@arijitAD lgtm, just one minor comment

File: x/market/client/cli/tx.go [hunk 2]
```diff
@@ -210,7 +210,7 @@ func cmdLeaseWithdraw(key string) *cobra.Command {
 	}
 
 	flags.AddTxFlagsToCmd(cmd)
-	AddLeaseIDFlags(cmd.Flags(), dcli.DeploymentIDOptionNoOwner(true))
```
Comment (boz) [inline x/market/client/cli/tx.go:213]: Didn't we just add these configuration options @troian?

Comment (troian) [reply x/market/client/cli/tx.go:213]: In this case they should be used by `Req` but not `Add`

**Comment by arijitAD** [comment]
> @arijitAD lgtm, just one minor comment

Verified that bid command takes owner flag as optional
Screenshot 2021-07-14 at 7 57 23 PM

**Review by troian — APPROVED** [review]

**Comment by dmikey** [comment]
Please update the original comment of this post with detail about the change.

**Comment by troian** [comment]
@dmikey merges are fine within PR lifecycle though I prefer rebase as well.
We do rebase, squash followed by merge when change is ready to go into the master

**Comment by troian** [comment]
@arijitAD when you merge make sure `Squash and merge` is selected. Commit message will look like
```
fix(cli): make owner flag optional. #1314
```

image

**Comment by dmikey** [comment]
The purpose of rebasing in PR is so that merge conflicts don't become owned by someone else trying to do the merge.

If the engineer pushing the changes rebases frequently, then merge conflicts virtually disappear, and the OG Engineer doesn't need to babysit merges.

**Comment by troian** [comment]
Frequent rebases won't prevent conflicts if two person work on different tasks within same files. It is still up to developer to resolve them anyways.

**Pull request merged**